src/components/site-form.tsx

```diff
@@ -254,6 +278,50 @@ export const SiteForm = ( {
 					<span className="font-semibold">{ __( 'Site name' ) }</span>
 					<TextControlComponent onChange={ setSiteName } value={ siteName }></TextControlComponent>
 				</label>
+				{ setUseCustomDomain && setCustomDomain && (
+					<div className="flex items-center gap-2 mb-6">
+						<input
+							type="checkbox"
+							id="use-custom-domain"
+							checked={ useCustomDomain }
+							onChange={ ( e ) => {
+								setUseCustomDomain( e.target.checked );
+								setCustomDomain( generateDomainFromSiteName( siteName ) );
+							} }
+						/>
+						<label htmlFor="use-custom-domain">{ __( 'Use custom domain' ) }</label>
+						<span className="text-a8c-gray-50 text-xs ml-2">
+							{ __( 'This will require your password to authorize changes to the hosts file.' ) }
+						</span>
+					</div>
+				) }
+
+				{ useCustomDomain && setCustomDomain && (
+					<div className="flex flex-col gap-2 mb-6">
+						<label htmlFor="custom-domain" className="font-semibold">
+							{ __( 'Domain name' ) }
+						</label>
+						<TextControlComponent
+							id="custom-domain"
+							value={ customDomain ?? '' }
+							onChange={ setCustomDomain }
+							placeholder="my-site.wp.local"
+						/>
+						{ customDomainError && (
+							<div className="text-red-500 text-xs">{ customDomainError }</div>
+						) }
```

Comment:
We usually use some kind of a `Notice` component for error messages like this.

Reply:
Didn't use a notice but I've used whatever they were using before.